### ISSUE TITLE
hash remains in wantlist even after adding and pinning

### ISSUE THREAD

**Issue by slothbag** [ipfs/go-ipfs#2859]
I have added a custom merkledag object using `object add` and yet `ipfs bitswap wantlist` still shows the hash as wanted.  I also tried pinning it and it still remains as wanted.

The content is 
```
{"Links":[{"Name":"newMS","Hash":"QmcnC9yF2riaf6fXhCe5N4MzaBgaqBK39JiJbSW62QNJFG","Size":0},{"Name":"op_1_ad","Hash":"QmSXFCMTVfg72a1bzQho8crm4sdmtEQ3nHz53ycnSJFGKA","Size":0}],"Data":""}
```
The hash is `QmVF1CxH4AhzpEswsMNge57M85dmzYpSvLX9TwsYqbt5zo`

Probably helps to know that my IPFS node has no peer connections (deliberate)

The IPFS node also contains all the child hashes too, and I tried recursive and not recursive, makes no difference.

**Comment by whyrusleeping** [comment]
Are you able to reproduce this?

**Comment by slothbag** [comment]
Well, the daemon was running, the content could be returned using object get but it remained in the wantlist.

I tried to add another custom object and it worked and did not appear in the wantlist.

I just restarted the daemon and now the original object is no longer appearing in the wantlist.

I'm not sure what steps lead to it getting stuck, it's no longer stuck after restart.  So no, I can't reproduce.
